File: app/src/main/java/com/kruma/kanta/service/IAperturaUsuario.java
package com.kruma.kanta.service;

import com.kruma.core.service.entity.ServiceCallback;
import com.kruma.core.util.common.List;
import com.kruma.core.util.common.ProcessResult;
import com.kruma.kanta.entidad.AperturaUsuario;

import retrofit.http.Body;
import retrofit.http.GET;
import retrofit.http.POST;
import retrofit.http.Path;

/**
 * Created by Jordy on 19/04/2017.
 */

public interface IAperturaUsuario {



    @POST("/Insertar")
    void Insertar(
            @Body com.kruma.kanta.entidad.AperturaUsuario AperturaUsuario,
            ServiceCallback<ProcessResult> obj_pCallBack);

    @GET("/ListarAperturaUsuario/{IdLocal}/{IdApertura}")
    void ListarAperturaUsuario(
            @Path("IdLocal") String str_pIdLocal,
            @Path("IdApertura") String str_pIdApertura,
            ServiceCallback<List<AperturaUsuario>> obj_pCallBack);

    @POST("/Modificar")
    void Modificar(
            @Body com.kruma.kanta.entidad.AperturaUsuario AperturaUsuario,
            ServiceCallback<ProcessResult> obj_pCallBack);


    @GET("/Obtener/{IdLocal}/{IdApertura}/{IdAperturaUsuario}/{IdUsuario}")
    void Obtener(
            @Path("IdLocal") String str_pIdLocal,
            @Path("IdApertura") String str_pIdApertura,
            @Path("IdAperturaUsuario") String str_pIdAperturaUsuario,
            @Path("IdUsuario") String str_pUsuario,
            ServiceCallback<AperturaUsuario> obj_pCallBack);



    @GET("/ObtenerEstadoAperturaUsuario/{IdLocal}/{IdApertura}/{IdAperturaUsuario}/{IdUsuario}")
    void ObtenerEstadoAperturaUsuario(
            @Path("IdLocal") String str_pIdLocal,
            @Path("IdApertura") String str_pIdApertura,
            @Path("IdAperturaUsuario") String str_pIdAperturaUsuario,
            @Path("IdUsuario") String str_pUsuario,
            ServiceCallback<AperturaUsuario> obj_pCallBack);


    @POST("/ModificarAperturaCancion")
    void ModificarAperturaCancion(
            @Body com.kruma.kanta.entidad.AperturaCancion AperturaCancion,
            ServiceCallback<ProcessResult> obj_pCallBack);
}

File: KantaPe_Web/Kruma.KantaPe/Kruma.KantaPe.Service/App_Code/Interface/IAperturaUsuario.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Runtime.Serialization;
using System.ServiceModel;
using System.ServiceModel.Web;
using System.Text;

namespace Kruma.KantaPe.Service.Interface
{
    [ServiceContract]
    public interface IAperturaUsuario
    {
       
        [OperationContract]
        [WebInvoke(
        Method = "POST",
        RequestFormat = WebMessageFormat.Json,
        ResponseFormat = WebMessageFormat.Json,
        BodyStyle = WebMessageBodyStyle.Bare,
        UriTemplate = "Insertar")]
        Kruma.KantaPe.Service.Entity.ServiceResult<Kruma.Core.Util.Common.ProcessResult>
            Insertar(Kruma.KantaPe.Entidad.AperturaUsuario AperturaUsuario);

        [OperationContract]
        [WebInvoke(
       Method = "GET",
       ResponseFormat = WebMessageFormat.Json,
       BodyStyle = WebMessageBodyStyle.Bare,
       UriTemplate = "ListarAperturaUsuario/{IdLocal}/{IdApertura}")]
        Kruma.KantaPe.Service.Entity.ServiceResult<
           Kruma.Core.Util.Common.List<Kruma.KantaPe.Entidad.AperturaUsuario>> ListarAperturaUsuario(
                   string IdLocal,
                   string IdApertura
           );

        [OperationContract]
        [WebInvoke(
              Method = "POST",
              RequestFormat = WebMessageFormat.Json,
              ResponseFormat = WebMessageFormat.Json,
              BodyStyle = WebMessageBodyStyle.Bare,
              UriTemplate = "ModificarAperturaCancion")]
        Kruma.KantaPe.Service.Entity.ServiceResult<Kruma.Core.Util.Common.ProcessResult>
                  ModificarAperturaCancion(Kruma.KantaPe.Entidad.AperturaCancion AperturaCancion);

        [OperationContract]
        [WebInvoke(
        Method = "POST",
        RequestFormat = WebMessageFormat.Json,
        ResponseFormat = WebMessageFormat.Json,
        BodyStyle = WebMessageBodyStyle.Bare,
        UriTemplate = "Modificar")]
        Kruma.KantaPe.Service.Entity.ServiceResult<Kruma.Core.Util.Common.ProcessResult>
            Modificar(Kruma.KantaPe.Entidad.AperturaUsuario AperturaUsuario);

        [OperationContract]
        [WebInvoke(
        Method = "GET",
        ResponseFormat = WebMessageFormat.Json,
        BodyStyle = WebMessageBodyStyle.Bare,
        UriTemplate = "Obtener/{IdLocal}/{IdApertura}/{IdAperturaUsuario}/{IdUsuario}")]
        Kruma.KantaPe.Service.Entity.ServiceResult<Kruma.KantaPe.Entidad.AperturaUsuario> Obtener(string IdLocal,
                   string IdApertura, string IdAperturaUsuario, string IdUsuario);

        [OperationContract]
        [WebInvoke(
        Method = "GET",
        ResponseFormat = WebMessageFormat.Json,
        BodyStyle = WebMessageBodyStyle.Bare,
        UriTemplate = "ObtenerEstadoAperturaUsuario/{IdLocal}/{IdApertura}/{IdAperturaUsuario}/{IdUsuario}")]
        Kruma.KantaPe.Service.Entity.ServiceResult<Kruma.KantaPe.Entidad.AperturaUsuario> ObtenerEstadoAperturaUsuario(string IdLocal,
                   string IdApertura, string IdAperturaUsuario, string IdUsuario);
    }
}
